### PR TITLE
Float sliders

### PR DESCRIPTION
**Pull request by Hperigo**
Hi there. 
thanks for the work with this library. It's super handy!

this pull-request implements a couple of different inputs:
1. DragInt
2. DragFloat
3. SliderFloat

### CODE REVIEW

File: imguiWrapper.cpp [hunk 1]
```diff
@@ -244,6 +244,21 @@ IggBool iggSliderInt(char const *label, int *value, int minValue, int maxValue,
    return ImGui::SliderInt(label, value, minValue, maxValue, format) ? 1 : 0;
 }
 
+IggBool iggDragInt( char const *label, int *value, int speed, int min, int max, const char* format)
```
Comment (dertseha) [inline imguiWrapper.cpp]: Please have the order be "char const *format" - I'm quite picky about that ;)
See also other occurence in iggDragFloat() and header.

**Comment by dertseha** [comment]
Hello there and thank you for your contribution!
Apart from the comment about specifier ordering, please have a look at the Go comments for the functions. Almost every one has an issue: either the wrong name of the function (the one documented, or the one called) and/or mistakes in the call list of the idiomatic forwarders.

(I also wonder why the linter wouldn't pick up those commenting issues)

**Comment by Hperigo** [comment]
hey there, thank you for taking the time to review the code! I'm kinda new to go, and the comments issues just slipped through  me.

one question. The DragFloat option in ImGui has a default value in the `min/max` parameter set to 0 making possible for the widget go to infinity. How would you approach that in go? 
make min/max float pointers? Or just write another function?

File: imgui.go [hunk 2]
```diff
@@ -360,6 +360,60 @@ func SliderInt(label string, value *int32, min, max int32) bool {
 	return SliderIntV(label, value, min, max, "%d")
 }
 
+// DragInt calls DragIntV(label, value, min, max, "%%d").
```
Comment (dertseha) [inline imgui.go]: format string is "%d"

File: imgui.go [hunk 3]
```diff
@@ -360,6 +360,60 @@ func SliderInt(label string, value *int32, min, max int32) bool {
...
+	labelArg, labelFin := wrapString(label)
+	defer labelFin()
+	valueArg, valueFin := wrapFloat(value)
+	defer valueFin()
+	formatArg, formatFin := wrapString(format)
+	defer formatFin()
+	return C.iggSliderFloat(labelArg, valueArg, C.float(min), C.float(max), formatArg) != 0
+}
+
+// SliderFloat calls SliderIntV(label, value, min, max, "%d").
```
Comment (dertseha) [inline imgui.go]: format string is "%f"

File: imgui.go [hunk 4]
```diff
@@ -360,6 +360,60 @@ func SliderInt(label string, value *int32, min, max int32) bool {
...
+func SliderFloat(label string, value *float32, min, max float32) bool {
+	return SliderFloatV(label, value, min, max, "%f")
+}
+
+// DragFloat calls DragFloatV(label, value, min, max, "%0.3f").
```
Comment (dertseha) [inline imgui.go]: speed parameter is missing

File: imgui.go (hunk 2 above)
Comment (dertseha) [reply imgui.go]: speed parameter is missing

**Comment by dertseha** [comment]
I just realized, the Drag*() functions of ImGui have in their idiomatic variant only the label and value parameters - speed, min, max, and the like are only in the verbose variant.
The intention of the "idiomatic" functions is to require only the parameter the C++ variant would require at minimum. All the defaulted parameters should only be in the verbose variant.

So, it should be e.g.
```
DragFloat(label string, value *float32) bool
DragFloatV(label String, value *float32, speed, min, max float32, format string, power float32) bool
```

As for your question, min/max=0  will be considered in the implementation as being without limits, which is an explicit behaviour, so that's OK. This is even documented in the header (past the first DragFloat function).

I'm sorry I don't have much time to make a thorough review these days thanks to busy times.

**Comment by Hperigo** [comment]
A ok, apologies for the delay awensering I got busy at work. Feel free to do changes in a different branch. I'll have some time this week and can work on that also. 

Another question: 
Imgui has some handy functions for vectors, example:  `ImGui::DragFloat2`, `ImGui::DragFloat3`, etc...

Is there any plan on integrating this lib with mathgl? 
[https://github.com/go-gl/mathgl/tree/master/mgl32](https://github.com/go-gl/mathgl/tree/master/mgl32)

If not, should the `DragFloat2` just accept the x,y value like this?
```DragFloat(label string, x, y *float32) bool ```

**Comment by dertseha** [comment]
All right, I'll have a go at importing this some time this week.

Apart from that, adding the DragFloatX variants should probably be done with either `[]float32` or `*[X]float32` parameter types (I'm tending towards the latter). This avoids adding a dependency, while staying compatible with mgl.

**Comment by dertseha** [comment]
(To complete the spam: I'm sorry for that, GitHub responded always with "can't comment right now", so, obviously I have to re-try several times)

**Pull request merged**